dali/pipeline/data/buffer.h

```diff
@@ -307,7 +326,45 @@ class DLL_PUBLIC Buffer {
     return shares_data_;
   }
 
-  DISABLE_COPY_MOVE_ASSIGN(Buffer);
+  /**
+   * @brief Set another Buffer as the backing memory for this Buffer.
+   *
+   * Current Buffer will be marked as sharing data, and reallocation of memory will be
+   * prohibited until reset() is called.
+   */
+  inline void set_backing_allocation(const Buffer<Backend> &other) {
+    type_ = other.type_;
+    data_ = other.data_;
+    allocate_ = {};
+    size_ = other.size_;
+    shares_data_ = true;
+    num_bytes_ = other.num_bytes_;
```

Comment:
How about this?
```suggestion
    num_bytes_ = other.num_bytes_;
    shares_data_ = num_bytes_ > 0;
```
Do we want the buffer to be set to "sharing" even if it doesn't really share anything?

Reply:
Hmm, make sense, but should we allow sharing (basically setting some allocation for this buffer), that is empty and does not result in sharing anything?

Reply:
Done

Reply:
Not done? I can still see hard `shares_data_ = true;`  there...

Reply:
Argh, I thought I did it.